### ISSUE TITLE
Fips packages should be pinned or held to avoid fips-breaking upgrades

### ISSUE THREAD

**Issue by panlinux** [CanonicalLtd/ubuntu-advantage-script#61]
After enabling FIPS, several packages are replaced by their FIPS counterparts. These come from a special FIPS PPA provided by Canonical.

Normal system upgrades, however, could replace these FIPS packages with the non-FIPS ones, thus rendering the installation uncertified.

It sounds reasonable that the ubuntu-advantage script, after having enabled fips and installed these extra needed packages, should somehow mark these as having a higher priority than the normal ubuntu repositories. This could be achieved either via "holds", or by pinning the fips PPA at a higher priority.

The PPA pinning sounds more attractive at first glance because it would rank any package that comes from that PPA as having a higher priority, so it's future proof.

**Comment by albertodonato** [comment]
I don't think we can pin packages, otherwise those won't ever get upgrades from the FIPS repo.
Adding an apt preference to make the PPA higher priority should work, though.
